### PR TITLE
add hu.getTargetDistance()

### PR DESCRIPTION
**Pull request by Goober5000**
Add a scripting function to return the distance as seen on the HUD.

### CODE REVIEW

File: code/scripting/api/libs/hud.cpp [hunk 1]
```diff
@@ -237,6 +240,29 @@ ADE_FUNC(flashTargetBox, l_HUD, "enumeration section, [number duration_in_millis
 	return ADE_RETURN_NIL;
 }
 
+ADE_FUNC(getTargetDistance, l_HUD, "object targetee, [vector targeter_pos]", "Returns the distance as displayed on the HUD, that is, the distance from a position to the bounding box of a target.  If targeter_pos is nil, the function will use the player's position.", "number", "The distance, or nil if invalid")
```
Comment (wookieejedi) [inline code/scripting/api/libs/hud.cpp:243]: Thoughts on changing this to `targeter_position` in both cases? Seems like, in general, that full name descriptions are a bit easier to read and more common in the scripting variable names.

Comment (Goober5000) [reply code/scripting/api/libs/hud.cpp:243]: Yeah that makes sense.

Comment (Goober5000) [reply code/scripting/api/libs/hud.cpp]: Well, in the documentation anyway.  In the code, I kept it as-is to be consistent with other usages.

Comment (wookieejedi) [reply code/scripting/api/libs/hud.cpp]: Ah yeah just the documentation was what I was noting.

Comment (Goober5000) [reply code/scripting/api/libs/hud.cpp]: Ok.  Got the other one in the documentation too.

**Review by wookieejedi — APPROVED** [review]

File: code/scripting/api/libs/hud.cpp [hunk 2]
```diff
@@ -237,6 +240,29 @@ ADE_FUNC(flashTargetBox, l_HUD, "enumeration section, [number duration_in_millis
...
+		if (Player_obj)
+			targeter_pos = &Player_obj->pos;
+		else
+		{
+			Warning(LOCATION, "Player_obj is NULL; cannot assign position!");
+			targeter_pos = &vmd_zero_vector;
+		}
+	}
+
+	auto dist = hud_find_target_distance(targetee->objp, targeter_pos);
```
Comment (asarium) [inline code/scripting/api/libs/hud.cpp:262]: `targetee` may be invalid so in order to avoid crashes, that should be checked before calling this.

**Review by asarium — CHANGES_REQUESTED** [review]

File: code/scripting/api/libs/hud.cpp (hunk 2 above)
Comment (Goober5000) [reply code/scripting/api/libs/hud.cpp:265]: Good point.  Updated.

Comment (asarium) [reply code/scripting/api/libs/hud.cpp:265]: You also need to check for an invalid handle since `objp` may be invalid otherwise.

Comment (Goober5000) [reply code/scripting/api/libs/hud.cpp]: Missed that.  Updated again.

**Review by asarium — APPROVED** [review]

**Pull request merged**